### ISSUE TITLE
[CI] DiskDisruptionIT.testGlobalCheckpointIsSafe UncaughtExceptionError

### ISSUE THREAD

**Issue by ywangd** [elastic/elasticsearch#68188]
**Build scan**:
https://gradle-enterprise.elastic.co/s/zyyqfcbgefxc2

**Repro line**:
./gradlew ':server:internalClusterTest' --tests "org.elasticsearch.discovery.DiskDisruptionIT.testGlobalCheckpointIsSafe" -Dtests.seed=4AC417739D6AD299 -Dtests.security.manager=true -Dtests.locale=es-AR -Dtests.timezone=Etc/GMT+4 -Druntime.java=8

**Reproduces locally?**:
No

**Applicable branches**:
7.11

**Failure history**:

[build-stats](https://build-stats.elastic.co/app/kibana#/discover?_g=(refreshInterval:(pause:!t,value:0),time:(from:now-90d,mode:quick,to:now))&_a=(columns:!(_source),filters:!(('$state':(store:appState),meta:(alias:!n,disabled:!f,index:e58bf320-7efd-11e8-bf69-63c8ef516157,key:class,negate:!f,params:(query:org.elasticsearch.discovery.DiskDisruptionIT,type:phrase),type:phrase,value:org.elasticsearch.discovery.DiskDisruptionIT),query:(match:(class:(query:org.elasticsearch.discovery.DiskDisruptionIT,type:phrase)))),('$state':(store:appState),meta:(alias:!n,disabled:!f,index:e58bf320-7efd-11e8-bf69-63c8ef516157,key:test,negate:!f,params:(query:testGlobalCheckpointIsSafe,type:phrase),type:phrase,value:testGlobalCheckpointIsSafe),query:(match:(test:(query:testGlobalCheckpointIsSafe,type:phrase)))),('$state':(store:appState),meta:(alias:!n,disabled:!f,index:e58bf320-7efd-11e8-bf69-63c8ef516157,key:type,negate:!f,params:(query:com.carrotsearch.randomizedtesting.UncaughtExceptionError,type:phrase),type:phrase,value:com.carrotsearch.randomizedtesting.UncaughtExceptionError),query:(match:(type:(query:com.carrotsearch.randomizedtesting.UncaughtExceptionError,type:phrase))))),index:e58bf320-7efd-11e8-bf69-63c8ef516157,interval:auto,query:(language:lucene,query:testGlobalCheckpointIsSafe),sort:!(time,desc)))
It failed only once in last 90 days. However the failure message looks suspicious to me and suggests there maybe a minor bug somewhere. Because it reminds me that we had a few subtle bugs related to calling `listener.onResponse` or `listener.onFailure` multiple times. I am aware that this failure looks similar to #58021 on the surface. But the underlying error message is quite different. So I am raising a new issue instead of re-openning the old one.

**Failure excerpt**:
```
org.elasticsearch.discovery.DiskDisruptionIT > testGlobalCheckpointIsSafe FAILED
    com.carrotsearch.randomizedtesting.UncaughtExceptionError: Captured an uncaught exception in thread: Thread[id=3824, name=elasticsearch[node_t0][generic][T#2], state=RUNNABLE, group=TGRP-DiskDisruptionIT]
        at __randomizedtesting.SeedInfo.seed([4AC417739D6AD299:76E700D797F92ADE]:0)

        Caused by:
        java.lang.AssertionError: IndexingPressure is adjusted twice
            at __randomizedtesting.SeedInfo.seed([4AC417739D6AD299]:0)
            at org.elasticsearch.index.IndexingPressure.lambda$wrapReleasable$0(IndexingPressure.java:71)
            at org.elasticsearch.action.ActionListener$5.onFailure(ActionListener.java:303)
            at org.elasticsearch.action.support.replication.TransportReplicationAction$AsyncPrimaryAction.onFailure(TransportReplicationAction.java:446)
            at org.elasticsearch.action.support.replication.TransportReplicationAction$AsyncPrimaryAction.handleException(TransportReplicationAction.java:440)
            at org.elasticsearch.action.support.replication.TransportReplicationAction$AsyncPrimaryAction.lambda$runWithPrimaryShardReference$3(TransportReplicationAction.java:425)
            at org.elasticsearch.action.ActionListener$1.onFailure(ActionListener.java:136)
            at org.elasticsearch.action.ActionListener$MappedActionListener.onFailure(ActionListener.java:97)
            at org.elasticsearch.action.support.replication.ReplicationOperation.finishAsFailed(ReplicationOperation.java:353)
            at org.elasticsearch.action.support.replication.ReplicationOperation.onNoLongerPrimary(ReplicationOperation.java:298)
            at org.elasticsearch.action.support.replication.ReplicationOperation.access$1100(ReplicationOperation.java:57)
            at org.elasticsearch.action.support.replication.ReplicationOperation$2.lambda$onFailure$2(ReplicationOperation.java:229)
            at org.elasticsearch.action.ActionListener$1.onFailure(ActionListener.java:136)
            at org.elasticsearch.action.ActionListener.onFailure(ActionListener.java:247)
            at org.elasticsearch.transport.TransportRequestDeduplicator$CompositeListener.onCompleted(TransportRequestDeduplicator.java:103)
            at org.elasticsearch.transport.TransportRequestDeduplicator$CompositeListener.onFailure(TransportRequestDeduplicator.java:117)
            at org.elasticsearch.cluster.action.shard.ShardStateAction$2.onClusterServiceClose(ShardStateAction.java:251)
            at org.elasticsearch.cluster.ClusterStateObserver$ContextPreservingListener.onClusterServiceClose(ClusterStateObserver.java:328)
            at org.elasticsearch.cluster.ClusterStateObserver$ObserverClusterStateListener.onClose(ClusterStateObserver.java:237)
            at org.elasticsearch.cluster.service.ClusterApplierService.addTimeoutListener(ClusterApplierService.java:263)
            at org.elasticsearch.cluster.ClusterStateObserver.waitForNextChange(ClusterStateObserver.java:176)
            at org.elasticsearch.cluster.ClusterStateObserver.waitForNextChange(ClusterStateObserver.java:120)
            at org.elasticsearch.cluster.action.shard.ShardStateAction.waitForNewMasterAndRetry(ShardStateAction.java:239)
            at org.elasticsearch.cluster.action.shard.ShardStateAction$1.handleException(ShardStateAction.java:169)
            at org.elasticsearch.transport.TransportService.sendRequest(TransportService.java:698)
            at org.elasticsearch.transport.TransportService.sendRequest(TransportService.java:686)
            at org.elasticsearch.cluster.action.shard.ShardStateAction.sendShardAction(ShardStateAction.java:159)
            at org.elasticsearch.cluster.action.shard.ShardStateAction.lambda$remoteShardFailed$2(ShardStateAction.java:210)
            at org.elasticsearch.transport.TransportRequestDeduplicator.executeOnce(TransportRequestDeduplicator.java:50)
            at org.elasticsearch.cluster.action.shard.ShardStateAction.remoteShardFailed(ShardStateAction.java:208)
            at org.elasticsearch.action.support.replication.TransportWriteAction$WriteActionReplicasProxy.failShardIfNeeded(TransportWriteAction.java:441)
            at org.elasticsearch.action.support.replication.ReplicationOperation$2.onFailure(ReplicationOperation.java:228)
            at org.elasticsearch.action.support.RetryableAction.cancel(RetryableAction.java:87)
            at org.elasticsearch.action.support.replication.PendingReplicationActions.lambda$cancelActions$0(PendingReplicationActions.java:112)
            at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
            at java.util.concurrent.ConcurrentHashMap$KeySpliterator.forEachRemaining(ConcurrentHashMap.java:3527)
            at java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:580)
            at java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:270)
            at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1384)
            at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:482)
            at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:472)
            at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)
            at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)
            at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
            at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:418)
            at org.elasticsearch.action.support.replication.PendingReplicationActions.lambda$cancelActions$1(PendingReplicationActions.java:112)
            at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:684)
            at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
            at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
            at java.lang.Thread.run(Thread.java:748)
```

**Comment by dnhatn** [comment]
Duplicate of #68049
